app/models/project.rb

```diff
@@ -125,6 +125,10 @@ def last_deploy_by_group(before_time)
     releases.map { |group_id, deploys| [ group_id, deploys.sort_by(&:updated_at).last ] }.to_h
   end
 
+  def deploy_groups
+    stages.uniq.map(&:deploy_groups)
```

Comment:
-> no need for the .uniq

also this gives a array with arrays ... I'd expect a flat array ... missing tests ... `stages.flat_map(&:deploy_groups).uniq`